File: solutions/ruby/raindrops/1/raindrops.rb
=begin
Write your code for the 'Raindrops' exercise in this file. Make the tests in
`raindrops_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/raindrops` directory.
=end

class Raindrops
  F = {3=>"Pling",5=>"Plang",7=>"Plong"}
  def self.convert(num)
    val = ""
    F.each do |factor,value|
      if num % factor == 0
        val += value
      end
    end
  val.empty? ? num.to_s : val
  end
end

File: solutions/go/raindrops/1/raindrops.go
package raindrops

import (    
    "fmt"
    "bytes"
)

func Convert(number int) string {
    var buffer bytes.Buffer
    if number % 3 == 0 {
        buffer.WriteString("Pling")
	}
	if number % 5 == 0 {
    	buffer.WriteString("Plang")
    }
	if number % 7 == 0 {
    	buffer.WriteString("Plong")
    }
	if buffer.Len() == 0 {
    	return fmt.Sprint(number)
    }
	return buffer.String()
}
